pennylane/math/quantum.py

```diff
@@ -733,6 +733,80 @@ def _compute_mutual_info(
     return vn_entropy_1 + vn_entropy_2 - vn_entropy_12
 
 
+# pylint: disable=too-many-arguments
+def vn_entanglement_entropy(
+    state, indices0, indices1, base=None, check_state=False, c_dtype="complex128"
+):
+    r"""Compute the Von Neumann entanglement entropy between two subsystems in a given state.
+
+    .. math::
+
+        S(\rho_A) = -\text{Tr}[\rho_A \log \rho_A] = -\text{Tr}[\rho_B \log \rho_B] = S(\rho_B)
+
+    where :math:`S` is the von Neumann entropy, and :math:`\rho_A = \text{Tr}_B[\rho_{AB}]` and
+    :math:`\rho_B = \text{Tr}_A[\rho_{AB}]` are the reduced density matrices for each partition.
+
+    The Von Neumann entanglement entropy is a measure of the degree of quantum entanglement between
+    two subsystems constituting a pure bipartite quantum state. The entropy of entanglement is the
+    Von Neumann entropy of the reduced density matrix for any of the subsystems. If it is non-zero,
+    it indicates the two subsystems are entangled.
+
+    Each state must be given as a density matrix. To find the mutual information given
+    a pure state, call :func:`~.math.dm_from_state_vector` first.
+
+    Args:
+        state (tensor_like): ``(2**N)`` state vector or ``(2**N, 2**N)`` density matrix.
+        indices0 (list[int]): List of indices in the first subsystem.
+        indices1 (list[int]): List of indices in the second subsystem.

```

Comment:
Should `indices0 | indices1 = {0, 1, 2, ..., N}`?

Reply:
Not necessarily, the two subsystems don't have to cover the fully system.

Reply:
Great! What prompts my question is that it's not clear in that page what `indices0` or `indices1` should in principle be.

Reply:
I would say that they are indices of the wires that constitutes each sub system, not the wire labels, but the indices of the wires.

Reply:
Agreed, we just never say that for $n$ wires we expect our indices to be subsets of $\\{1, \ldots, n \\}$